test/src/unit-capi-nullable.cc

```diff
@@ -555,7 +537,7 @@ void NullableArrayFx::do_2d_nullable_test(
     const tiledb_layout_t cell_order,
     const tiledb_layout_t tile_order,
     const tiledb_layout_t write_order) {
-  const string array_name = "2d_nullable_array";
+  const string array_path = temp_dir_.value().path() + "2d_nullable_array";
```

Comment:
It seems better to keep the previous `array_name` argument and to compute the path internally.

All the statements with `FILE_TEMP_DIR + array_name` can be converted to `array_path(array_name)` calls, with `array_path` being a new member function of the fixture.